python/shape.cpp

```diff
@@ -40,12 +40,33 @@ template <class T> void bind_concatenate(py::module &m) {
 template <class T> void bind_reshape(pybind11::module &mod) {
   mod.def(
       "reshape",
-      [](const T &self, const std::vector<Dim> &labels,
-         const py::tuple &shape) {
-        Dimensions dims(labels, shape.cast<std::vector<scipp::index>>());
-        return reshape(self, dims);
+      [](const T &self, const py::dict &dims) {
```

Comment:
I think it's not too bad having to use a `zip` if you want to use params from another object. Alternatively, you can also just use `.sizes` from another object, and you no longer need the `zip`.
Actually, for people who don't know too much about python (or maps in programming in general), I would say that the fact that some maps don't preserve order is more surprising. All people who were beginning with people that i spoke to just expected the dimension to be preserved and didn't stop to think that it could actually be any other way.
Finally, I think using a dict is more Scipp-like, where again you don't have to remember any dimension order between `dims` and `shape`, you directly specify "i want 10 elements in dim `x`".
But maybe the signature is a point we should discuss with the rest of the scipp team.
I also thought it was ok to change it because it seems pretty unused throughout the docs, and wouldn't break many things.